contracts/RewardsRunning.sol

```diff
@@ -0,0 +1,569 @@
+        __Ownable_init();
+
+        blockLimit = _blockLimit;
+        boosterProjectShare = _boosterProjectShare;
+        alphaNumerator = 1;
+        alphaDenominator = 3;
+        settings = _settings;
+    }
+
+    /**
+     * @notice update the settings
+     * @param _settings settings contract address
+     */
+    function setSettings(ISettings _settings) external onlyOwner {
+        settings = _settings;
+    }
+
+    /**
+     * @notice update the project shares
+     * @param projects all project ids
+     * @param shares all project shares
+     * @param deleted deleted projects
+     */
+    function setProjectShare(uint256[] calldata projects, uint256[] calldata shares, uint256[] calldata deleted) external onlyOwner {
+        require(projects.length == shares.length, 'RR003');
+
+        // check latest actived era
+        uint256 era = IEraManager(settings.getContractAddress(SQContracts.EraManager)).safeUpdateAndGetEra();
+        require(era == latestActivedEra + 1, 'RR010');
+
+        uint256 totalShare = 0;
+        for(uint256 i = 0; i < projects.length; i++) {
+            require(shares[i] > 0, 'RR011');
+            basicProjectShares[projects[i]] = shares[i];
+            totalShare += shares[i];
+        }
+        require(totalShare == 100, 'RR004');
+
+        for(uint256 i = 0; i < deleted.length; i++) {
+            delete basicProjectShares[deleted[i]];
+        }
+    }
+
+    /**
+     * @notice add booster project staking
+     * @param projectId the project id
+     * @param amount the added amount
+     */
+    function addProjectStaking(uint256 projectId, uint256 amount) external {
+        // check latest actived era
+        uint256 era = IEraManager(settings.getContractAddress(SQContracts.EraManager)).safeUpdateAndGetEra();
+        require(era == latestActivedEra + 1, 'RR010');
+
+        IERC20(settings.getContractAddress(SQContracts.SQToken)).safeTransferFrom(msg.sender, address(this), amount);
+
+        BoosterProject storage project = boosterProjectStakings[projectId];
+        project.amount += amount;
+        project.lastTime = block.number;
+        if (project.owner == address(0)) {
+            project.owner = msg.sender;
+        }
+        totalBoosterProjectStaking += amount;
+    }
+
+    /**
+     * @notice add booster project staking
+     * @param projectId the project id
+     * @param amount the added amount
+     */
+    function subProjectStaking(uint256 projectId, uint256 amount) external {
+        // check latest actived era
+        uint256 era = IEraManager(settings.getContractAddress(SQContracts.EraManager)).safeUpdateAndGetEra();
+        require(era == latestActivedEra + 1, 'RR010');
+
+        BoosterProject storage project = boosterProjectStakings[projectId];
+        require(project.lastTime + blockLimit < block.number, 'RR020');
+
+        IERC20(settings.getContractAddress(SQContracts.SQToken)).safeTransferFrom(address(this), project.owner, amount);
```

Comment:
why only owner can remove? I think anyone should be able to contribute to any project's booster pool